common/models/templates/src/trie-model.ts

```diff
@@ -121,7 +121,7 @@
 
     predict(transform: Transform, context: Context): Distribution<Suggestion> {
       // Special-case the empty buffer/transform: return the top suggestions.
-      if (!transform.insert && context.startOfBuffer && context.endOfBuffer) {
+      if (!transform.insert && !context.left && !context.right && context.startOfBuffer && context.endOfBuffer) {
```

Comment:
Can you explain the point of this to me?

Reply:
Sure.  Note the original comment:  `// Special case - the empty buffer/transform`.  (Slight edit for readability)

The original code matched more scenarios than the comment would indicate.  For example:

```
let context = {
  left: "The quick brown",
  right: " fox jumped over the lazy dog.",
  startOfBuffer: true,
  endOfBuffer: true
}
```

is a perfectly valid context.

For said context, a word-break operation should note that the current word-token is "brown", as the caret is effectively after the 'n'.  These additional checks allow us to detect that the word _has_ context and is not from an empty buffer, so we can make use of the existing prefix `"brown"` instead of disregarding it and starting new suggestions from scratch.

Reply:
I guess I don't understand what `startOfBuffer` and `endOfBuffer` are doing then!